packages/driver/src/cy/commands/waiting.ts

```diff
@@ -270,8 +281,51 @@ export default (Commands, Cypress, cy, state) => {
     })
   }
 
+  Cypress.primaryOriginCommunicator.on('wait:for:xhr', ({ args: [str, options] }, originPolicy) => {
+    options.isCrossOriginSpecBridge = true
+    waitString(null, str, options).then((responses) => {
+      // remove props that aren't serializable (these aren't used by consumers so are okay to remove)
+      const omitFn = (response) => _.omit(response, ['subscriptions', 'setLogFlag'])
+      const preprocessedResponses = _.isArray(responses) ? responses.map(omitFn) : omitFn(responses)
+
+      Cypress.primaryOriginCommunicator.toSpecBridge(originPolicy, 'wait:for:xhr:end', { responses: preprocessedResponses })
```

Comment:
I wonder if we should bake this handling into the `communicator` itself as it already handles specific serialization for some namespaced events. That way the pre/post processing is just contained to the communicator.

Reply:
The custom serialization was removed. [1b15780](https://github.com/cypress-io/cypress/pull/21579/commits/1b15780a25ec1035bef5f4a915b3b1d060d1abfc)